plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/RootResolvedTypes.java

```diff
@@ -131,7 +131,7 @@ protected void addTypeDiagnostic(XExpression expression, /* @Nullable */ TypeDat
 				if (expectedType != null) {
 					if (!expectedType.isPrimitiveVoid()) {
 						int conformanceFlags = getConformanceFlags(typeData, false);
-						if (!((conformanceFlags & ConformanceFlags.SUCCESS) != 0)) {
+						if ((conformanceFlags & ConformanceFlags.SUCCESS) == 0 || (conformanceFlags & ConformanceFlags.UNKNOWN_TYPE_PARTICIPATED) != 0) {
```

Comment:
— Bar.xtend —
```
class Bar {
  def void bar(Unknown it) {}
}
```

This file has an error for an unresolved type.

— Foo.xtend —
```
class Foo {
  def void foo(Bar bar) {
    bar.bar(notify)
  }
}
```

This file does not have any errors, but compilation fails, since notify of void type. Adding a conformance problem for the notify call will fix the issue.